src/fmu/pem/pem_functions/effective_pressure.py

```diff
@@ -13,81 +16,171 @@
+            fipnum_values = input_num_string_to_list(
+                zone_region.fipnum, actual_fipnum_values
+            )
+
+            # Build combined mask for all FIPNUMs in this zone using vectorized
+            # operation
+            zone_mask = np.isin(fipnum_data, fipnum_values) & ~fipnum_mask
+
+            # Check if this zone uses patchy cement model
+            is_patchy_cement = zone_region.model.model_name == RPMType.PATCHY_CEMENT
+
+            if is_patchy_cement:
+                # Get cement fraction and density for patchy cement model
+                cement_fraction = zone_region.model.parameters.cement_fraction
+                cement_density = rock_matrix.minerals[rock_matrix.cement].density
+            else:
+                cement_fraction = None
+                cement_density = None
+
+            # Calculate bulk density for this zone
+            zone_bulk_density = estimate_bulk_density(
+                porosity=np.ma.masked_where(~zone_mask, sim_init.poro),
+                fluid_density=[np.ma.masked_where(~zone_mask, fl_dens)],
+                mineral_density=np.ma.masked_where(~zone_mask, matrix_props.density),
+                patchy_cement=is_patchy_cement,
+                cement_fraction=cement_fraction,
+                cement_density=cement_density,
             )
-            for (sim_date, dens) in zip(sim_rst, bulk_density)
-        ]
-    else:  # overburden_pressure.type == 'trend':
-        eff_pres = [
-            estimate_effective_pressure(
-                formation_pressure=sim_date.pressure * 1.0e5,
-                bulk_density=dens,  # type: ignore
-                reference_overburden_pressure=overburden_pressure_from_trend(
-                    inter=overburden_pressure.intercept,
-                    grad=overburden_pressure.gradient,
-                    depth=sim_init.depth,
-                ),
+
+            # Merge zone bulk density into full grid
+            bulk_dens_grid[zone_mask] = zone_bulk_density[0][zone_mask]
+
+        bulk_density_list.append(bulk_dens_grid)
+
+    bulk_density = bulk_density_list
+
+    # Calculate overburden pressure per zone (time-independent)
+    # Initialize overburden pressure grid
+    overburden_pressure_grid = np.ma.masked_array(
+        np.full(sim_init.depth.shape, np.nan, dtype=float), mask=fipnum_mask
+    )
+
+    for zone_idx, zone in enumerate(overburden_pressure):
+        # Get all FIPNUM values for this zone using input_num_string_to_list
+        fipnum_values = input_num_string_to_list(zone.fipnum, actual_fipnum_values)  # type: ignore
+
+        # Build combined mask for all FIPNUMs in this zone using vectorized operation
+        mask_cells = np.isin(fipnum_data, fipnum_values) & ~fipnum_mask
+
+        if zone.type == OverburdenPressureTypes.CONSTANT:
+            # Constant overburden pressure for this zone (convert from bar to Pa)
+            overburden_pressure_grid[mask_cells] = zone.value * 1.0e5
```

Comment:
It could be more readable to have constants like `1.0e5` defined in the top as e.g.

`BAR_TO_PASCAL = 1.0e5`